algorithms/src/r1cs/test_constraint_checker.rs

```diff
@@ -73,9 +73,9 @@ impl<F: Field> TestConstraintChecker<F> {
 impl<F: Field> ConstraintSystem<F> for TestConstraintChecker<F> {
     type Root = Self;
 
-    fn alloc<Fn, A, AR>(&mut self, _annotation: A, f: Fn) -> Result<Variable, SynthesisError>
+    fn alloc<Fn, A, AR>(&mut self, _annotation: A, f: Fn) -> Result<Variable, anyhow::Error>
```

Comment:
Conditioned on adding `use anyhow::Result;` to the file (the same applies to other places; all can be found with the **regexp** search for `Result<.*, anyhow::Error>,`)
```suggestion
    fn alloc<Fn, A, AR>(&mut self, _annotation: A, f: Fn) -> Result<Variable>
```